department-of-veterans-affairs/vets-api
# [API-40280] Appoint the POA for the dependent claimant (v1)

## Summary

- Add dependent claimant POA assignment service. We can now assign POAs to dependents using the v1 /2122 endpoint. 
- If the v1 /2122 request has the `claimant` attribute and the feature flag is enabled, call the dependent claimant POA assignment service (and assign the POA to the dependent).

## Related issue(s)

[API-40280](https://jira.devops.va.gov/browse/API-40280)

## Testing done


**Steps to test**
1. Find a Veteran in VBMS (test env).
1. Send a POST request to `/services/claims/v1/forms/2122`.
    To get the below screenshot, I sent the following request:
    ```
    curl --location 'http://localhost:3000/services/claims/v1/forms/2122' \
    --header 'X-VA-SSN: 796378881' \
    --header 'X-VA-First-Name: Jesse' \
    --header 'X-VA-Last-Name: Gray' \
    --header 'X-VA-Birth-Date: 1954-12-15' \
    --header 'Content-Type: application/json' \
    --header 'Authorization: ••••••' \
    --data '{
        "data": {
            "type": "form/21-22",
            "attributes": {
                "serviceOrganization": {
                    "poaCode": "003"
                },
                "claimant": {
                    "firstName": "Merry",
                    "lastName": "Test",
                    "address": {
                        "numberAndStreet": "123 anystreet",
                        "city": "anytown",
                        "country": "USA",
                        "zipFirstFive": "12345"
                    },
                    "relationship": "Spouse"
                }
            }
        }
    }'
   ```
1. Observe a successful response.
1. Verify the dependentʼs POA was updated in VBMS.

**Troubleshooting**

- Is the `lighthouse_claims_api_poa_dependent_claimants` feature flag enabled?
- Does the POA under test exist in the local database and BGS?

## Screenshots
**Before**
![before](https://github.com/user-attachments/assets/b4f01e96-0291-4c12-b82a-59a948875902)

**After**
![after](https://github.com/user-attachments/assets/9ff5c7e7-091a-4d12-8cbf-9e9d3a3ff7ac)

## What areas of the site does it impact?

- v1 Power of Attorney submit request

## Acceptance criteria

- [x]  I updated|added unit tests and integration tests for each feature (if applicable).

## Requested Feedback

Our methods for checking BGSʼ responses rely on the string messages we receive. Open to suggestions on how to make these checks more robust.
